commands/args.go

```diff
@@ -191,90 +189,29 @@ func NewArgs(args []string) *Args {
 	}
 }
 
-func slurpGlobalFlags(args *[]string, globalFlags *[]string, noop *bool) {
-	var (
-		globalFlagSet flag.FlagSet
-
-		configParam      mapValue = make(mapValue)
-		paginate         bool
-		noPaginate       bool
-		noReplaceObjects bool
-		bare             bool
-		version          bool
-		help             bool
-
-		execPath string
-		gitDir   string
-		workTree string
-	)
-
-	globalFlagSet.BoolVarP(noop, "noop", "", false, "")
-	globalFlagSet.VarP(configParam, "", "c", "")
-	globalFlagSet.BoolVarP(&paginate, "paginate", "p", false, "")
-	globalFlagSet.BoolVarP(&noPaginate, "no-pager", "", false, "")
-	globalFlagSet.BoolVarP(&noReplaceObjects, "no-replace-objects", "", false, "")
-	globalFlagSet.BoolVarP(&bare, "bare", "", false, "")
-	globalFlagSet.BoolVarP(&version, "version", "", false, "")
-	globalFlagSet.BoolVarP(&help, "help", "", false, "")
-
-	globalFlagSet.StringVarP(&execPath, "exec-path", "", "", "")
-	globalFlagSet.StringVarP(&gitDir, "git-dir", "", "", "")
-	globalFlagSet.StringVarP(&workTree, "work-tree", "", "", "")
-
-	globalFlagSet.SetOutput(ioutil.Discard)
-	globalFlagSet.Init("hub", flag.ContinueOnError)
-
-	aa := make([]string, 0)
-	err := globalFlagSet.Parse(*args)
-	if err == nil {
-		aa = globalFlagSet.Args()
-	} else {
-		aa = *args
-	}
-
-	// manipulate global flags
-	if version {
-		aa = append([]string{"version"}, aa...)
-	}
-
-	if help {
-		aa = append([]string{"help"}, aa...)
-	}
-
-	for k, v := range configParam {
-		*globalFlags = append(*globalFlags, "-c")
-		*globalFlags = append(*globalFlags, fmt.Sprintf("%s=%s", k, v))
-	}
-
-	if paginate {
-		*globalFlags = append(*globalFlags, "--paginate")
-	}
-
-	if noPaginate {
-		*globalFlags = append(*globalFlags, "--no-pager")
-	}
-
-	if noReplaceObjects {
-		*globalFlags = append(*globalFlags, "--no-replace-objects")
-	}
-
-	if bare {
-		*globalFlags = append(*globalFlags, "--bare")
-	}
-
-	if execPath != "" {
-		*globalFlags = append(*globalFlags, "--exec-path", execPath)
-	}
+func slurpGlobalFlags(args *[]string, globalFlags *[]string) {
+	slurpNextValue := false
+	commandIndex := 0
 
-	if gitDir != "" {
-		*globalFlags = append(*globalFlags, "--git-dir", gitDir)
+	for i, arg := range *args {
+		if slurpNextValue {
+			commandIndex = i + 1
+			slurpNextValue = false
+		} else if arg == "--version" || arg == "--help" || !strings.HasPrefix(arg, "-") {
+			break
+		} else {
+			commandIndex = i + 1
+			if arg == "-c" || arg == "-C" {
+				slurpNextValue = true
+			}
+		}
 	}
 
-	if workTree != "" {
-		*globalFlags = append(*globalFlags, "--work-tree", workTree)
+	if commandIndex > 0 {
+		aa := *args
+		*globalFlags = aa[0:commandIndex]
```

Comment:
I wonder what happens if there're multiple `-c` and they're not consequent to each other, e.g., `hub -c a=b --git-dir=X -c c=d`. It doesn't seem to work. Could you add a test for that?